client/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -125,15 +126,16 @@ object RetryPolicy {
     GatewayTimeout
   )
 
-  /** Returns true if the request method is idempotent and the result is
-    * either a throwable or has one of the `RetriableStatuses`.
+  /** Returns true if (the request method is idempotent or request contains Idempotency-Key header)
+    * and the result is either a throwable or has one of the `RetriableStatuses`.
     *
     * Caution: if the request body is effectful, the effects will be
     * run twice.  The most common symptom of this will be resubmitting
     * an idempotent request.
     */
   def defaultRetriable[F[_]](req: Request[F], result: Either[Throwable, Response[F]]): Boolean =
-    req.method.isIdempotent && isErrorOrRetriableStatus(result)
+    (req.method.isIdempotent || req.headers.get[`Idempotency-Key`].isDefined) &&
```

Comment:
Regarding the header, `idempotency-key`, must the HTTP Request's Method be a `POST`, @vglushak @rossabaker?

https://tools.ietf.org/html/draft-idempotency-header-00#section-2.2 notes:

> The idempotency key that is supplied as part of every "POST" request
MUST be unique and can not be reused with another request with a
different request payload.

In other words, should this instead be:

```scala
(req.method.isIdempotent || (req.headers.get[`Idempotency-Key`].isDefined && req.method === Method.Post)
```

?

Reply:
Oops, I hadn't seen this before I approved.  This is a good question.  I don't see any reason that the header couldn't be applied to a `PATCH` request.  And I think if the client cared enough care to provide this header, it's with the intent to treat the request as idempotent.  The spec isn't absolute, but I think this implementation is correct as is.